contract_payment_mode/models/contract.py

```diff
@@ -9,15 +9,19 @@ class ContractContract(models.Model):
         string="Payment Mode",
         domain=[("payment_type", "=", "inbound")],
         index=True,
+        compute="_compute_payment_mode_id",
+        store=True,
+        readonly=False,
     )
 
-    @api.onchange("partner_id")
-    def on_change_partner_id(self):
-        partner = self.with_company(self.company_id).partner_id
-        if self.contract_type == "purchase":
-            self.payment_mode_id = partner.supplier_payment_mode_id.id
-        else:
-            self.payment_mode_id = partner.customer_payment_mode_id.id
+    @api.depends("partner_id")
```

Comment:
Shouldn't it also depend on the 'contract_type' field?

Reply:
Yes, it should, changed